sdks/python/apache_beam/ml/transforms/handlers.py

```diff
@@ -83,20 +84,53 @@
 tft_process_handler_output_type = typing.Union[beam.Row, Dict[str, np.ndarray]]
 
 
+# alternatie: Use a single class for both encoding and decoding and
+# use beam.Map() instead of DoFns?
+class _EncodeDict(beam.DoFn):
+  """
+  Encode a dictionary into bytes and pass it along with the original element
+  using a temporary key.
+
+  Internal use only. No backward compatibility guarantees.
+  """
+  def __init__(self, exclude_columns=None):
+    self._exclude_columns = exclude_columns
+
+  def process(self, element: Dict[str, Any]):
+    data_to_encode = element.copy()
+    for key in self._exclude_columns:
+      if key in data_to_encode:
+        del data_to_encode[key]
+
+    bytes = pickler.dumps(data_to_encode)
```

Comment:
It might be more efficient to use type-aware cythonized coder. Performance difference can be tested in a microbenchmark.

We might be able to convert elements to Beam Row, and use RowCoder. Seeing something similar in:
https://github.com/apache/beam/blob/a221f98a5f46be985afcb98a65fcec3b46b81f92/sdks/python/apache_beam/transforms/external.py#L127-L132

Can we expect the schema of the elements in the dict the same ? do we have type information from the elements or would need to infer them? 
 
cc: @robertwb who might have a better idea

Reply:
> For the unused elements, we won't know what the schema of the elements.

From mltransform docs elements end up being Rows:

> To define a data processing transformation by using MLTransform, create instances of data processing transforms with columns as input parameters. The data in the specified columns is transformed and outputted to the beam.Row object.

do we infer the types later then?

Reply:
Only the data for the. columns that are provided for transformation. For these columns using TFT, we infer the schema